cohere/client.py

```diff
@@ -35,13 +35,28 @@ def __init__(self,
                  version: str = None,
                  num_workers: int = 64,
                  request_dict: dict = {},
-                 check_api_key: bool = True) -> None:
+                 check_api_key: bool = True,
+                 client_name: str = None) -> None:
+        """
+        Initialize the client.
+        Args:
+           * api_key (str): Your API key.
+           * version (str): API version to use. Will use cohere.COHERE_VERSION by default.
+           * num_workers (int): Maximal number of threads for parallelized calls.
+           * request_dict (dict): Additional parameters for calls to requests.post
+           * check_api_key (bool): Whether to check the api key for validity on initialization.
+           * client_name (str): A string to identify your application for analytics purposes.
```

Comment:
Should we indicate that it's for analytics purposes _on the cohere end?_ We don't expect to give external access to this data, do we?

Developers who need to do tracking for analytics can do so in the application which encapsulates API calls made via the SDK.

Reply:
I think it does not make a lot of sense to allow it per call. If you really need to (multiple sub-applications?), you could simply instantiate multiple clients.

Reply:
Fair enough.

Reply:
Wait, this is responding to my other comment, no? What about the point about the docstring?

Reply:
Don't really have a strong opinion on that one, but added 'internal'.